flink-connector-mysql-cdc/src/test/java/com/alibaba/ververica/cdc/connectors/mysql/MySQLSourceTest.java

```diff
@@ -744,10 +909,72 @@ public void go() throws Exception {
         return Tuple2.of(offsetFile, offsetPos);
     }
 
+    private static Properties createDebeziumProperties(boolean useLegacyImplementation) {
+        Properties debeziumProps = new Properties();
+        if (useLegacyImplementation) {
+            debeziumProps.put("internal.implementation", "legacy");
+            // check legacy mysql record type
+            debeziumProps.put("transforms", "snapshotasinsert");
+            debeziumProps.put(
+                    "transforms.snapshotasinsert.type",
+                    "io.debezium.connector.mysql.transforms.ReadToInsertEvent");
```

Comment:
Are the configurations also required when using legacy mode? I mean do users also need to set them when using legacy mode.

Reply:
yes